scvi/distributions/_negative_binomial.py

```diff
@@ -91,14 +93,27 @@ def log_nb_positive(x: torch.Tensor, mu: torch.Tensor, theta: torch.Tensor, eps=
 
     log_theta_mu_eps = torch.log(theta + mu + eps)
 
+    if is_sparse:
+        nonzero_x = x[x > 0]
+        terms_for_nonzero = (
+            theta * (torch.log(theta + eps) - log_theta_mu_eps)
+            + nonzero_x * (torch.log(mu + eps) - log_theta_mu_eps)
+            + torch.lgamma(nonzero_x + theta)
+            - torch.lgamma(theta)
+            - torch.lgamma(nonzero_x + 1)
```

Comment:
need to mask `mu`, `log_theta_mu_eps` with the nonzero mask.

the term `torch.lgamma(nonzero_x + theta)` might be trickier because theta has not been previously broadcast. Right here its shape is [1, n_genes], and the shape of nonzero_x will be a 1d tensor with all nonzero terms. So maybe theta should be initially broadcast (see Pyro code)